### DISCUSSION TITLE
About unique id of Inverter

### DISCUSSION THREAD

**Discussion by sofkaski** [davidrapan/ha-solarman#214] [General]
I noticed from the home assistant log the following message while updating to the latest version: 

> 2024-10-14 11:28:57.265 ERROR (MainThread) [homeassistant.config_entries] Config entry 'Inverter' from integration solarman has an invalid unique_id 'nnnnnnnnnn', please create a bug report at ...

('nnnnnnnnnnn' being my inverter serial number).

I looked at `/homeassistant/.storage/core.config_entries` with the following `jq` spell:

> jq '.data.entries[] | select(.unique_id == \<my inverter serial\>)' /homeassistant/.storage/core.config_entries

The result shows the unique id like: `"unique_id":  1234567890`  (instead of 1234567890 there is my serial number) .

So I quess that the issue is that unique id is a number instead of a string. @davidrapan: Has there been a change in the format/type of the unique id at some point in development? There are no migrations defined and that would hint it has not changed? I have defined my configuration quite a long time ago.

I did not create a bug report yet in the case this is not an issue with a newly created configuration.

**Comment by davidrapan** [comment]
If it's device_unique_id then yes. There was a change and was made in [v24.09.06](https://github.com/davidrapan/ha-solarman/releases/tag/v24.09.06)

The format is: solarman_1234567890 (1234567890 is serial number of the stick logger)

**Comment by sofkaski** [comment]
I patched the id in `core.config_entries` and `core.device_registry` to the new format and now the error message on restart is gone.
Thanks @davidrapan for the hint!

I suppose that in the future we should version the config entry and make migrations as described [here](https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#config-entry-migration) in the change cases like this.

**Reply by davidrapan** [reply]
The thing is that device_unique_id was not utilized at all before that so I'm nor really sure it would help but anyway I found out about the config versioning and migrations after the changes so... yeah!
